src/android/com/outsystems/firebase/analytics/validator/OSFANLEventParameterValidator.kt

```diff
@@ -85,7 +88,22 @@ class OSFANLEventParameterValidator private constructor(
             hasCurrency = hasCurrency || key == CURRENCY.json
 
             parameterKeySet.add(key)
-            result.putAny(key, value)
+
+            try {
+                // value, tax, and shipping are actually decimal values, not strings
+                if (key == VALUE.json || key == TAX.json || key == SHIPPING.json) {
+                    result.putAny(key, value.toDouble()) // can throw NumberFormatException
+                } else {
+                    result.putAny(key, value)
+                }
+            } catch (e: NumberFormatException) {
+                Log.d(
+                    "OSFANLEventParameterValidator",
+                    "Tried to convert non-number to double. Exception: ${e.message}" 
```

Comment:
would this be a good use case for the new `warning` structure? instead of just logging native side I mean

Reply:
Yes, that's a good idea! Although I think the best approach here would be to create backlog to make this a warning, since we currently aren't using the Warning structure in this plugin and this fix is kind of urgent.

What do you think?

Reply:
agreed!